### DISCUSSION TITLE
insert into无法和with as一起使用.

### DISCUSSION THREAD

**Discussion by howardyan93** [apache/doris#22760] [A - General / Q&A]
目前使用还在使用旧版**1.0.0-preview。**

在某个查询的过程中发现insert into和with as无法一起使用。

试过:
```
insert into tb1
with 
  tb2 as (select * from tb3 where condition)
select * from tb2;
```

和
```
with 
  tb2 as (select * from tb3 where condition)
insert into tb1
select * from tb2;
```

都不行， 看了下文档说insert into **with label**, 所以解析不第一种表达方式，不知道有什么解决办法没有。

**Comment by LemonLiTree** [comment]
First insert into must be before with as.
Secondly, this problem has been solved in the new optimizer in version 2.0.0, you need "set enable_nereids_dml=true";

**Reply by heichong** [reply]
2.0.2版本，设置参数“set enable_nereids_dml=true”后，是否支持insert overwrite ？按你说的设置后，发现insert into 可以，但是 insert overwrite 不行

**Reply by mark-triker** [reply]
now i use version 2.12,insert overwrite is available

**Reply by parkour99** [reply]
> First insert into must be before with as. Secondly, this problem has been solved in the new optimizer in version 2.0.0, you need "set enable_nereids_dml=true";

doris 3.0 do not support with as (sql ), you can try it again now

**Comment by ygq010718** [comment]
这个问题，怎么解决啊，实现insert overwrite

**Comment by LemonLiTree** [comment]
not support insert overwrite

**Comment by ixzc** [comment]
we have supported insert overwrite
https://doris.apache.org/zh-CN/docs/sql-manual/sql-statements/Data-Manipulation-Statements/Manipulation/INSERT-OVERWRITE
